npm/vue/src/index.ts

```diff
@@ -129,7 +129,10 @@ export function mount<V extends {}>(
     __vccOpts: any
   },
   options?: MountingOptions<any> & Record<string, any>
-): Cypress.Chainable<VueWrapper<ComponentPublicInstance<V>>>
+): Cypress.Chainable<{
+  wrapper: VueWrapper<ComponentPublicInstance<V>>
+  component: VueWrapper<ComponentPublicInstance<V>>['vm']
+}>
```

Comment:
Still working on my Typescript, but would it make sense to introduce a generic return type like:
```
type MountReturn<C> = {
  wrapper: VueWrapper<C>
  component: C
}
```
to use for the return of all the mount overloads?

_Result_
```
export function mount<V extends {}>(
  originalComponent: {
    new (...args: any[]): V
    __vccOpts: any
  },
  options?: MountingOptions<any> & Record<string, any>
): Cypress.Chainable<MountReturn<V>>
```

Would that work for all the overload cases?